### PR TITLE
vtgate table schema tracking to use GetSchema rpc

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR uses the new GetSchema rpc for retrieving table schema. This RPC was added in v17 and used in v18.

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/12967

## Checklist

-   [X] Tests were added or are not required
-   [X] Did the new or modified tests pass consistently locally and on the CI

### CODE REVIEW

**Comment by harshit-gangal** [comment]
Here the test `TestSelectScatterOrderByVarChar` does not have weight_string in the query but same test with stream is still having weight_string in the query `TestStreamSelectScatterOrderBy`. In the second test, the column is infact an integer.
Is this a miss in the planner?

File: go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go [hunk 1]
```diff
@@ -86,15 +83,10 @@ func TestMain(m *testing.M) {
...
-		clusterInstance.VtGatePlannerVersion = planbuilder.Gen4
-		clusterInstance.VtTabletExtraArgs = []string{"--queryserver-config-schema-change-signal",
-			"--queryserver-config-schema-change-signal-interval", "0.1",
-			"--queryserver-config-strict-table-acl",
-			"--queryserver-config-acl-exempt-acl", "userData1",
-			"--table-acl-config", "dummy.json"}
+		clusterInstance.VtGateExtraArgs = append(clusterInstance.VtGateExtraArgs,
+			"--schema_change_signal",
+			"--vschema_ddl_authorized_users", "%")
+		clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, "--queryserver-config-schema-change-signal")
```
Comment (GuptaManan100) [inline go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:89]: Some acl flags have also been dropped. Is that intentional?

File: go/test/endtoend/vtgate/schematracker/sharded_prs/st_sharded_test.go [hunk 2]
```diff
@@ -145,8 +145,8 @@ func TestMain(m *testing.M) {
 			sidecarDBName = sidecardb.DefaultName
 		}
 
-		clusterInstance.VtGateExtraArgs = []string{"--schema_change_signal", "--schema_change_signal_user", "userData1"}
-		clusterInstance.VtTabletExtraArgs = []string{"--queryserver-config-schema-change-signal", "--queryserver-config-schema-change-signal-interval", "5", "--queryserver-config-strict-table-acl", "--queryserver-config-acl-exempt-acl", "userData1", "--table-acl-config", "dummy.json"}
+		clusterInstance.VtGateExtraArgs = append(clusterInstance.VtGateExtraArgs, "--schema_change_signal")
+		clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, "--queryserver-config-schema-change-signal")
```
Comment (GuptaManan100) [inline go/test/endtoend/vtgate/schematracker/sharded_prs/st_sharded_test.go:149]: Same acl question

File: go/vt/vtgate/schema/tracker.go [hunk 3]
```diff
@@ -130,8 +102,17 @@ func (t *Tracker) loadTables(conn queryservice.QueryService, target *querypb.Tar
...
+	var numTables int
+	err := conn.GetSchema(t.ctx, target, querypb.SchemaTableType_TABLES, nil, func(schemaRes *querypb.GetSchemaResponse) error {
+		t.updateTables(target.Keyspace, schemaRes.TableDefinition)
+		numTables += len(schemaRes.TableDefinition)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, numTables)
```
Comment (GuptaManan100) [inline go/vt/vtgate/schema/tracker.go:115]: Here we are loading tables. I think the comment is wrong.

**Review by GuptaManan100 — APPROVED** [review]
Just one more thing I noticed, in the `tracker.go` file `aclErrorMessageLog` is now an unused constant and can be cleaned up

Rest looks good to me!

File: go/test/endtoend/vtgate/schematracker/restarttablet/schema_restart_test.go [hunk 4]
```diff
@@ -91,10 +90,7 @@ func TestMain(m *testing.M) {
 
 		// restart the tablet so that the schema.Engine gets a chance to start with existing schema
 		tablet := clusterInstance.Keyspaces[0].Shards[0].PrimaryTablet()
-		tablet.VttabletProcess.ExtraArgs = []string{
-			"--queryserver-config-schema-change-signal",
-			fmt.Sprintf("--queryserver-config-schema-change-signal-interval=%d", signalInterval),
-		}
+		tablet.VttabletProcess.ExtraArgs = append(tablet.VttabletProcess.ExtraArgs, "--queryserver-config-schema-change-signal")
```
Comment (frouioui) [inline go/test/endtoend/vtgate/schematracker/restarttablet/schema_restart_test.go:93]: We are no longer using the signal interval is that expected? If yes, we should remove the definition of `signalInterval` on line `45`.

**Review by frouioui — APPROVED** [review]
Approving to unblock progress and left one comment about unused variable.

File: go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go (hunk 1 above)
Comment (harshit-gangal) [reply go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:89]: It is not required anymore. I will add a deprecation note for it.

File: go/test/endtoend/vtgate/schematracker/sharded_prs/st_sharded_test.go (hunk 2 above)
Comment (harshit-gangal) [reply go/test/endtoend/vtgate/schematracker/sharded_prs/st_sharded_test.go:149]: https://github.com/vitessio/vitess/pull/13544#discussion_r1270564303

**Pull request merged**